appinventor/appengine/src/com/google/appinventor/client/explorer/dialogs/ProjectPropertiesDialogBox.java

```diff
@@ -0,0 +1,192 @@
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.uibinder.client.UiHandler;
+import com.google.gwt.user.client.ui.Button;
+import com.google.gwt.user.client.ui.DeckPanel;
+import com.google.gwt.user.client.ui.Label;
+import com.google.gwt.user.client.ui.ListBox;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import com.google.gwt.user.client.ui.Widget;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+
+/**
+ * A dialog for updating project properties that can be open from any screen
+ */
+public class ProjectPropertiesDialogBox { 
+
+    private static ProjectPropertiesDialogBoxUiBinder uiBinder =
+        GWT.create(ProjectPropertiesDialogBoxUiBinder.class);
+
+    interface ProjectPropertiesDialogBoxUiBinder extends UiBinder<Widget, ProjectPropertiesDialogBox> {
+    }
+
+    @UiField 
+    Dialog projectProperties;
+
+    @UiField
+    ListBox projectPropertyCategoryTitlePanel;
+
+    @UiField
+    DeckPanel propertiesDeckPanel;
+
+    @UiField
+    Button closeDialogBox;
+
+    /**
+     * List Of project properties category, which will be used to group properties in the dialog 
+     * properties category are : General, Theming, Publishing
+     */
+    private static final ArrayList<String> projectPropertyCategoryTitle = new ArrayList<String>() {{
+        add(Ode.MESSAGES.projectPropertyGeneralCategoryTitle());
+        add(Ode.MESSAGES.projectPropertyThemingCategoryTitle());
+        add(Ode.MESSAGES.projectPropertyPublishingCategoryTitle());
+    }};
+
+    /**
+     * Maps the project property category to List of EditableProperty which
+     * belongs to that particular project property category
+     */
+    private HashMap<String, ArrayList<EditableProperty>> categoryToProperties = new HashMap<>();
+
+    /* Object for stroing reference of project editor in which the dialog opned */
+    private YaProjectEditor projectEditor;
+
+    /* refers to the screen name in which dialog opend */
+    private String currentScreen = "";
+
+    /**
+     * Show the project property dialog
+     * 
+     * @param screenName name of the screen in which dialog needs to open
+     */
+    public void showDialog(String screenName) {
+        currentScreen = screenName;
+        Scheduler.get().scheduleDeferred(new Scheduler.ScheduledCommand() {
+            @Override
+            public void execute() {
+                projectProperties.center();
+            }
+        });   
+    }
+
+    public ProjectPropertiesDialogBox() {
+        uiBinder.createAndBindUi(this);
+        projectProperties.setAutoHideEnabled(false);
+        projectProperties.setModal(true);
+        projectProperties.setCaption(Ode.MESSAGES.projectPropertiesText());
+
+        // Get current instance of YaProjectEditor
+    	projectEditor = (YaProjectEditor)Ode.getInstance().getEditorManager().getOpenProjectEditor(
+            Ode.getInstance().getCurrentYoungAndroidProjectId());
+
```

Comment:
Since the constructor is called from YaProjectEditor, it seems like it would make more sense for the owner to pass a reference to itself via the constructor rather than taking this roundabout way through Ode.